packages/cli/src/commands/onboard/auth/github/index.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { OAuthApp } from '@octokit/oauth-app';
+import chalk from 'chalk';
+import * as fs from 'fs-extra';
+import inquirer from 'inquirer';
+import { Task } from '../../../../lib/tasks';
+import { updateConfigFile } from '../../config';
+import { APP_CONFIG_FILE, PATCH_FOLDER } from '../../files';
+import { patch } from '../patch';
+
+type Answers = {
+  clientSecret: string;
+  clientId: string;
+  hasEnterprise: boolean;
+  enterpriseInstanceUrl?: string;
+};
+
+const validateCredentials = async (clientId: string, clientSecret: string) => {
+  try {
+    const app = new OAuthApp({
+      clientId,
+      clientSecret,
+    });
+    await app.createToken({
+      code: '%NOT-VALID-CODE%',
+    });
+  } catch (error) {
+    // @octokit/request returns a error.response object when a request is rejected.
+    // We can check it to see what kind of error we received.
+
+    // If error.response is successful we can double-check that the error itself was due to the bad code.
+    // If that's the case then we can assume that the client id and secret exists as we otherwise would
+    // have gotten a 400/404.
+    if (
+      error.response.status !== 200 &&
+      error.response.data.error !== 'bad_verification_code'
+    ) {
+      throw new Error(`Validating GitHub Credentials failed.`);
+    }
+  }
+};
+
+const getConfig = (answers: Answers) => {
+  const { clientId, clientSecret, hasEnterprise, enterpriseInstanceUrl } =
+    answers;
+
+  return {
+    auth: {
+      providers: {
+        github: {
+          development: {
+            clientId,
+            clientSecret,
+            ...(hasEnterprise && {
+              enterpriseInstanceUrl,
+            }),
+          },
+        },
+      },
+    },
+  };
+};
+
+export const github = async () => {
```

Comment:
Would prefer if we avoid this much code in the index files, rather create a separate file and re-export. Or perhaps `onboard/auth/github.ts` could be enough in this case

Reply:
moved this (& `gitlab/index.ts`) to `providers/github.ts`